### ISSUE TITLE
labels are not honored by lb services

### ISSUE THREAD

**Issue by sangeethah** [rancherio/rancher#1325]
Steps to reproduce the problem:

Used the following docker-compose.yml and rancher-compose.yml  to create LB services with host affinity labels:

docker-compose.yml 
```
lb1234:
  ports:
  - 7777:80
  tty: true
  image: rancher/load-balancer-service
  labels:
    io.rancher.scheduler.affinity:host_label: lbhost=true
  links:
  - web1:web1
  stdin_open: true
web1:
  restart: 'no'
  labels:
    io.rancher.scheduler.affinity:host_label: lbhost=true
  tty: true
  image: sangeetha/testlbsd
  stdin_open: true
```
rancher-compose.yml 
```
lb1234:
  scale: 1
  load_balancer_config:
    description: lb1234
    name: lb1234 config
    health_check:
      interval: 2000
      unhealthy_threshold: 3
      request_line: ''
      healthy_threshold: 2
      response_timeout: 2000
web1:
  scale: 2
```

host affinity labels are not being saved as part of lb services and we see that the LB Agent gets deployed in host that do not have the specified host labels.


| 1071 | web1          |          5 | service             | c6116193-d153-4a62-9184-0ef38308e195 | NULL        | active   | 2015-06-25 16:39:23 | NULL                | NULL                | {"fields":{"scale":2,"launchConfig":{"imageUuid":"docker:sangeetha/testlbsd","labels":{"io.rancher.scheduler.affinity:host_label":"lbhost=true"},"logConfig":{},"networkMode":"managed","restartPolicy":{"name":"no"},"stdinOpen":true,"tty":true,"privileged":false,"publishAllPorts":false,"readOnly":false,"startOnCreate":true}}}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |            479 |

| 1072 | lb1234        |          5 | loadBalancerService | de5230fa-3593-4fa0-8449-1bdbabce754d | NULL        | active   | 2015-06-25 16:39:31 | NULL                | NULL                | {"fields":{"scale":1,"launchConfig":{"ports":["7777:80"],"networkMode":"managed","privileged":false,"publishAllPorts":false,"readOnly":false,"startOnCreate":true,"stdinOpen":false,"tty":false},"loadBalancerConfig":{"description":"lb1234","healthCheck":{"healthyThreshold":2,"interval":2000,"responseTimeout":2000,"unhealthyThreshold":3},"name":"lb1234 config"}}}                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  |            479 |

**Comment by alena1108** [comment]
Looks like rancher-compose bug. Just tested with Rancher API - labels are respected and set on the lb service as well as on the instances that are the part of this service.

**Comment by deniseschannon** [comment]
Verified fixed in v0.1.3 of Rancher Compose, which is now latest in Rancher.

**Comment by sangeethah** [comment]
Tested with rancher-compose version v0.1.3.

We are able to create Lb services with host affinity labels by setting labels in docker-compose.yml for Lb services

Lb service and the LB Agent created has the labels set.

| 1307 | lb1234        |          5 | loadBalancerService | c92abd7e-abe9-4441-a825-6546c9001c45 | NULL        | active   | 2015-06-25 21:35:06 | NULL                | NULL                | {"fields":{"scale":3,"launchConfig":{"labels":{"io.rancher.scheduler.affinity:host_label":"lbhost=true"},"ports":["7777:80"],"networkMode":"managed","privileged":false,"publishAllPorts":false,"readOnly":false,"startOnCreate":true,"stdinOpen":false,"tty":false},"loadBalancerConfig":{"description":"lb1234","healthCheck":{"healthyThreshold":2,"interval":2000,"responseTimeout":2000,"unhealthyThreshold":3},"name":"lb1234 config"}}}
